virttest/bootstrap.py

```diff
@@ -300,6 +301,115 @@ def create_guest_os_cfg(t_type):
               guest_os_cfg_path)
 
 
+def host_os_get_distro_name(options, detected, compat=False):
+    """
+    Gets the distro name, either from the command line or auto detection
+
+    :param options: parsed command line arguments results
+    :type options: :class:`argparse.Namespace`
+    :param detected: result of :class:`avocado.utils.distro.detect`
+    :type detected: :class:`avocado.utils.distro.LinuxDistro`
+    :param compat: If set, name is returned as uppercase or capitalized
+    :type compat: bool
+    """
+    if options.vt_host_distro_name:
+        return options.vt_host_distro_name
+    if compat:
+        if detected.name == 'rhel':
+            return 'RHEL'
+        elif detected.name == 'fedora':
+            return 'Fedora'
+    return detected.name
+
+
+def host_os_get_distro_version(options, detected, compat=False):
+    """
+    Gets the distro version, either from the command line or auto detection
+
+    :param options: parsed command line arguments results
+    :type options: :class:`argparse.Namespace`
+    :param detected: result of :class:`avocado.utils.distro.detect`
+    :type detected: :class:`avocado.utils.distro.LinuxDistro`
+    :param compat: If set, distro version with an "m" (as in major) prefix
+    :type compat: bool
+    """
+    if options.vt_host_distro_version:
+        version = options.vt_host_distro_version
+    else:
+        version = detected.version
+    if compat:
+        return "m%s" % version
+    else:
+        return version
+
+
+def host_os_get_distro_release(options, detected, compat=False):
+    """
+    Gets the distro release, either from the command line or auto detection
+
+    :param options: parsed command line arguments results
+    :type options: :class:`argparse.Namespace`
+    :param detected: result of :class:`avocado.utils.distro.detect`
+    :type detected: :class:`avocado.utils.distro.LinuxDistro`
+    :param compat: If set, distro version with an "u" (as in update) prefix
+    :type compat: bool
+    """
+    if options.vt_host_distro_release:
+        release = options.vt_host_distro_release
+    else:
+        release = detected.release
+    if compat:
+        return "u%s" % release
+    else:
+        return release
+
+
+def host_os_get_distro_arch(options, detected):
+    """
+    Gets the distro arch, either from the command line or auto detection
+
+    :param options: parsed command line arguments results
+    :type options: :class:`argparse.Namespace`
+    :param detected: result of :class:`avocado.utils.distro.detect`
+    :type detected: :class:`avocado.utils.distro.LinuxDistro`
+    """
+    if options.vt_host_distro_arch:
+        return options.vt_host_distro_arch
+    else:
+        return detected.arch
```

Comment:
Some places may have the restriction of guest architecture like `only|no x86_64`, so this will cause some potential problem if host's has the same value as guest's.